### ISSUE TITLE
Set up Cirrus CI

### ISSUE THREAD

**Issue by pbludov** [checkstyle/checkstyle#8855]
> We can append new CIs:
> Azure Pipelines
> https://cirrus-ci.org/features/

Cirrus CI supports Windows, it looks like we can move some appveyor builds to this CI.

_Originally posted by @pbludov in https://github.com/checkstyle/checkstyle/issues/8805#issuecomment-697166475_

**Comment by pbludov** [comment]
@romani please set up an account for Checkstyle at  https://cirrus-ci.org/features/

**Comment by pbludov** [comment]
We can move windows build for JDK 14/15 to this CI.

**Comment by strkkk** [comment]
is it a duplicate of #8852?

**Comment by pbludov** [comment]
> is it a duplicate of #8852?

Looks like it is. I'm focused on the Windows image though.

**Comment by romani** [comment]
I installed integration of this CI to github.
@pbludov , please send first PR with very very basic config to make sure it works, most likely first PR will not activate it as it will activated only after first merge.

**Comment by pbludov** [comment]
> @pbludov , please send first PR with very very basic config to make sure it works

It works. Let's decide which configurations should be kept in appveyor and which move to cirrus.

I suggest JDK 8, 11 (two last stable) in appveyor; JDK 15 (latest) in cirrus.

**Comment by romani** [comment]
I ok with suggestion, please do

**Comment by pbludov** [comment]
Master build failed:
```
C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\ci>call git fetch origin pull//head:pull/ 
fatal: invalid refspec 'pull//head:pull/'
```
There should be an `if`.

**Comment by romani** [comment]
Please send PR with badge to readme.md if they support it.

It is hard for me to see build failure in some CI if it doesn't have badge.

**Comment by pbludov** [comment]
The workaround for connection reset hits a bug in HttpClient:
https://issues.apache.org/jira/browse/HTTPCORE-634

The bug was fixed in `4.4.14`, but maven version plugin is still [uses version](https://www.mojohaus.org/versions-maven-plugin/dependencies.html) `4.4.11`

**Comment by pbludov** [comment]
The failed build
https://cirrus-ci.com/task/6587210826776576?command=test_site_without_verify

**Comment by pbludov** [comment]
Master build [succeeded](https://github.com/checkstyle/checkstyle/runs/1186603103)
